clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -183,9 +183,9 @@ public void assignFromSubscribed(Collection<TopicPartition> assignments) {
         Set<TopicPartition> newAssignment = new HashSet<>(assignments);
         removeAllLagSensors(newAssignment);
 
-        for (TopicPartition tp : assignments)
-            if (!this.subscription.contains(tp.topic()))
-                throw new IllegalArgumentException("Assigned partition " + tp + " for non-subscribed topic.");
+//        for (TopicPartition tp : assignments)
```

Comment:
I'm still pondering on it, the issue is that during rebalance streams leader can possibly assign internal topic partitions while the consumer’s not aware of the topic yet (since these topics are not created in Kafka), the above check will fail. This is actually not introduced in this topic as even before we could hit this issue but it was not exposed by tests.

I think in general this check is valid, and in that case maybe I have to get an extended consumer class as we discussed offline..